### DISCUSSION TITLE
OIDCWarden Feedbacks

### DISCUSSION THREAD

**Discussion by Timshel** [Timshel/OIDCWarden#1] [Announcements]
Hey,

Just a place to ask any question :).

**Comment by simonszu** [comment]
Is a merge to upstream planned sometimes or do i have to decide between Vaultwarden without OIDC or OIDCwarden?

**Reply by Timshel** [reply]
If I manage to obtain enough support yes I would reopen and support the PR. 
Note that the [PR](https://github.com/dani-garcia/vaultwarden/pull/3899) did not include all the work present in the version I distributed on docker: cf https://github.com/Timshel/vaultwarden/?tab=readme-ov-file#additionnal-features
So even if it was merged additional PR will then be needed to reach feature parity (If accepted).

**Reply by Timshel** [reply]
Forgot to update at the time but for those reading this, the PR was reopened a week later and the goal is still to upstream everything.

**Comment by ibotty** [comment]
I understand the motivation for the fork.  Can one just point an installation of your vaultwarden image to the OIDCwarden image?

**Reply by Timshel** [reply]
I tried to keep renaming to a minimum. The executable changed to `oidcwarden` and I removed some legacy compat in the start [script](https://github.com/Timshel/OIDCWarden/blob/main/docker/start.sh). But outside of it all env variable are the same.

I do not deploy it with the image, but the switch in the Playwright compose was transparent.
So proceed with caution, but I expect it should work without issues :).

**Comment by derfabianpeter** [comment]
Hi. Is there a way to contact you regarding professional financial support of your work?

**Reply by Timshel** [reply]
Edit: removed :).

**Reply by Timshel** [reply]
@derfabianpeter hey reaching here to confirm you received my email. Thanks.

**Comment by delta-whiplash** [comment]
Thanks for your amazing work
Am trying to setup a sso for my homelab and I want to use my k3s cluster, is a helm chart is planned ?

**Reply by Timshel** [reply]
Hey, I'm not using Kubernetes so not planned.
I believe there are many charts made for Vaultwarden which should work with minimum modifications.

**Comment by Camore5845** [comment]
Thank you for this amazing project!
I was able to add it to my organization, and I also use it personally.
It’s exactly what we needed! 😉

**Comment by MouaisTe** [comment]
Thank you so much for your contribution! I tried it with Authentik, and it worked perfectly.
However, I'm disappointed that we still have to use the master password to log in to the vault. I wanted to use this functionality mainly to avoid people in my organization having to remember more than one password. I understand that it's needed for encryption, but isn't there any other way to implement this functionality? Or if it's not possible, we can have the choice to replace the password by a PIN code to maintain encryption.

**Reply by Timshel** [reply]
Hey,
Since the goal is to use the clients maintained by Bitwarden the goal is to make the minimum of modifications possible.
There is an unlock with pin [feature](https://bitwarden.com/help/unlock-with-pin/), but it will not replace the master password.

**Comment by LeWimbes** [comment]
Hello Timshel and thanks for your amazing work.
Given the recent issues with the admin diagnostics page, I wanted to implement a regression test to prevent similar problems in the future.
Since you already built a Playwright setup in [PR #3899](https://github.com/dani-garcia/vaultwarden/pull/3899), would you be willing to move that setup into a separate PR? Having the test infrastructure merged sooner would let others begin adding tests while you continue with the SSO/OpenID work.

**Reply by Timshel** [reply]
Hey,
Last year I made a separate PR to add only the tests but after some initial feedback it just languished.
In the end I closed it when I created this fork to lower the work needed to maintain everything.

Since I don't think the situation changed much I don't plan to open it again for now.

**Comment by gjuuz** [comment]
Can someone explain what is the difference between this and vaultwarden i did not understand this 100%.

**Reply by Timshel** [reply]
At the moment almost none since there was a recent release of `Vaultwarden`, both project are built on top.

Main difference is the release cycle, the `Vaultwarden` project I distribute match the project release. Ex previous release was based `1.33.2` from Feb 9.
With this project I follow more closely the Bitwarden web client, which mean for more frequent release, tradeoff is they are not as well tested, but I'm expanding the Playwright tests.

In the long term I'll probably remove the `override` web-client distribution option from `Vaultwarden` (the one which remove the need to set an email) and the current logic of hiding/changing the state of the login buttons (base on `SSO_ENABLED` and `SSO_ONLY`) might be deemed too complex and would then be removed.

**Comment by bbglz0r** [comment]
Hello!

I'm a business analyst looking for a password manager for our team, and I've been evaluating various open-source solutions. I was particularly impressed by Vaultwarden, and when I discovered OIDCwarden, I was even more excited-since it addresses a major need for SSO. However, I've run into some difficulties, and I apologize in advance if this turns out to be a basic question.

Background
On my Windows 10 Pro laptop, I used Docker to spin up two containers: one running Keycloak and another running NGINX + OIDCwarden. I configured a realm in Keycloak and was able to log into OIDCwarden successfully.

The Issue
When creating organizations and configuring access in Keycloak, I can't get them to be created automatically on first login to OIDCwarden. To work around this, I manually created the organization in OIDCwarden, then created a Keycloak group (with the same name as the organization) and assigned the appropriate roles in Keycloak. Despite that, on my next login, I still don't receive access to the organization.

I'm almost certain I've missed or misconfigured something, but I can't pinpoint what it is.

I would greatly appreciate any help or pointers!
I've attached my .env file and a couple of screenshots from Keycloak.
[keycloak-docker-compose.txt](https://github.com/user-attachments/files/20612385/keycloak-docker-compose.txt)
[docker-compose.txt](https://github.com/user-attachments/files/20612386/docker-compose.txt)
[env.txt](https://github.com/user-attachments/files/20612387/env.txt)
![image](https://github.com/user-attachments/assets/68d1e98f-02e6-4943-a04d-785b9b12b88b)
![image](https://github.com/user-attachments/assets/e34185dd-0f10-45fe-b233-7046564efa74)
![image](https://github.com/user-attachments/assets/e4605885-9511-4fc7-960c-fd2c4fbfdc51)
![image](https://github.com/user-attachments/assets/f468b992-660c-4039-96f7-7757d034681f)

Many thanks to Timshel for this fork.

**Reply by Timshel** [reply]
Let's discuss it there: https://github.com/Timshel/OIDCWarden/issues/29 :)

**Reply by gjuuz** [reply]
Is it possible to sign in without master password when using this ?

**Reply by Timshel** [reply]
@gjuuz no

**Comment by cocoon** [comment]
Where do I get my "fee" "back"? 😁

![feeback](https://github.com/user-attachments/assets/38f93621-6fb6-422f-be4b-ae34de85bb29)

No I just wanted to say many thanks again for working on it, just deployed it today from scratch and found some nice things.

Great to see that the start.sh script already provides a way to run custom scripts, that allowed me to just insert a script that on startup runs:

`update-ca-certificates`

👍

Don't know if this already documented anywhere but this is a nice thing to add to a docker-compose file for example.

I simply added a volume for the ca-certificates and for /etc/oidcwarden.d:

```yaml
volumes:
     - ./bitwarden:/data:rw
     - ../ca-certificates:/usr/local/share/ca-certificates:ro
     - ./oidcwarden.d:/etc/oidcwarden.d:ro
```

and the script does nothing else than updating the certs at startup:

```
root@6a6ffbe40a76:/# cat /etc/oidcwarden.d/update-ca-certs.sh
update-ca-certificates
```
